### PR TITLE
added regex verification for CTB enum type

### LINKED ISSUE #13428: Enumeration caused Strapi to crash and wipe all data
### System information

- Node.js version: 16.15.0
- NPM version: 8.5.5
- Strapi version: 4.1.11
- Database: postgres
- Operating system: Ubunto 22.04

### Describe the bug

I was creating a new component in the Content-Type Builder. The component contained an Enum list with the following values:

- --16/9
- --4/5

Once i hit "save" Strapi crashed, giving a 504 error. When it restarted all the data was gone.

### Steps to reproduce the behavior

1. Go to Content-Type Builder
2. Click on Create new component
3. Select "Enumeration" as a field for the component
4. In the field "Values" enter: --16/9 and --4/5 on separate lines
5. Save the component
6. Crash - all data gone

### Expected behavior

I don't think putting " -- " in an Enum list should result in a crash and data wipe

Is there a way for me to get the data back?

### PR DESCRIPTION
**Pull request by omacovei**
Enumeration type in CTB was not tested if it starts with a special character

Fix #13428
Ready to be merged

I added a regex check that does not allow to create enum type if it starts with a special character different than '_'

Before see Issue #13428
Also it was seeing different strings as duplicates as in the printscreen below:

Screenshot 2022-06-07 at 16 29 44

After 
Screenshot 2022-06-08 at 13 47 05

### Tests run
Screenshot 2022-06-08 at 13 51 36

Screenshot 2022-06-08 at 13 47 59

### CODE REVIEW

File: packages/core/content-type-builder/admin/src/components/FormModal/attributes/types.js [hunk 1]
```diff
@@ -5,6 +5,7 @@ import getTrad from '../../../utils/getTrad';
 import getRelationType from '../../../utils/getRelationType';
 import toRegressedEnumValue from '../../../utils/toRegressedEnumValue';
 import startsWithANumber from '../../../utils/startsWithANumber';
+import matchRegex from '../../../utils/matchRegex';
```
Comment (gu-stav) [inline packages/core/content-type-builder/admin/src/components/FormModal/attributes/types.js:8]: I think you can remove this additional import and the file. I'll explain below why:

File: packages/core/content-type-builder/admin/src/components/FormModal/attributes/types.js [hunk 2]
```diff
@@ -172,7 +173,13 @@ const types = {
           name: 'doesNotStartWithANumber',
           message: getTrad('error.validation.enum-number'),
           test: values => !values.some(startsWithANumber),
+        })
+        .test({
+          name: 'matchesRegex',
+          message: getTrad('error.validation.match-regex'),
+          test: values => values.every(matchRegex),
```
Comment (gu-stav) [inline packages/core/content-type-builder/admin/src/components/FormModal/attributes/types.js:180]: The regex you need already exists, so there is no need to duplicate the knowledge about it:

```suggestion
          test: values => values.every(vENUM_REGEX.test),
```

What do you think?

**Review by gu-stav — COMMENTED** [review]
Thanks you for the PR. I've added a small comment and will need to verify the regex is valid and not too rigid.

**Comment by derrickmehaffy** [comment]
I think we should clarify that message to exactly what is allowed not just that it doesn't follow the regex.

File: packages/core/content-type-builder/admin/src/components/FormModal/attributes/types.js [hunk 3]
```diff
@@ -172,7 +172,13 @@ const types = {
           name: 'doesNotStartWithANumber',
           message: getTrad('error.validation.enum-number'),
           test: values => !values.some(startsWithANumber),
+        })
+        .test({
+          name: 'matchesRegex',
+          message: getTrad('error.validation.match-regex'),
```
Comment (gu-stav) [inline packages/core/content-type-builder/admin/src/components/FormModal/attributes/types.js:178]: To be more consistent with the other error names, could you please prefix the message:

```suggestion
          message: getTrad('error.validation.enum-match-regex'),
```

File: packages/core/content-type-builder/admin/src/translations/en.json [hunk 4]
```diff
@@ -60,6 +60,7 @@
   "error.contentTypeName.reserved-name": "This name cannot be used in your project as it might break other functionalities",
   "error.validation.enum-duplicate": "Duplicate values are not allowed",
   "error.validation.enum-empty-string": "Empty strings are not allowed",
+  "error.validation.match-regex": "The value does not match the regex.",
```
Comment (gu-stav) [inline packages/core/content-type-builder/admin/src/translations/en.json:63]:
```suggestion
  "error.validation.enum-match-regex": "At least one value contains invalid characters. Allowed are alphanumericals and _.",
```

**Review by gu-stav — COMMENTED** [review]
Apologies for the delay - I have been on vacation. Thanks for working on the PR - it looks almost good to me now. What do you think about applying @derrickmehaffy s suggestion to improve the error message?

**Comment by gu-stav** [comment]
Thank you again for opening the PR. We discussed it further yesterday and believe that it would be better to test on the regressed value instead. I've create another PR and included you as co-author: https://github.com/strapi/strapi/pull/13717 🙏🏼

**Pull request closed without merging**